src/app/components/feed/jh-main-feed-info.directive.js

```diff
@@ -27,12 +27,77 @@
       /* jshint: validthis */
       var vm = this;
       vm.name = name;
+      vm.makeNameEditable = makeNameEditable;
+      vm.leaveEditingMode = leaveEditingMode;
+      vm.editing = false; /* Name Editing mode */
+      vm.isEditing = isEditing;
+      vm.updateName = updateName;
+      vm.newName = null;
+      vm.keyDetect = keyDetect;
 
       function name() {
         var feed = FeedsService.findMain();
         var name = (feed !== null) ? feed.display : "";
         return name;
       }
+
+      /**
+       * Detects the key pressed by user
+       * @param {Number} - numeric keycode
+       */
+      function keyDetect(event) {
+        // if keypressed is 'Enter'
+        // then update the name to the current value in input box
+        console.log("This is detected keycode", event);
+        if (event.keyCode === 13) {
+          vm.updateName();
+          vm.leaveEditingMode();
+        // if keypressed is 'Escape'
+        // then leave the editing mode without changing name
+        } else if (event.keyCode === 27) {
+          vm.leaveEditingMode();
+        }
+      }
+      
```

Comment:
`keyDetect()` function listens for a keypress event in the input box. If `Enter` is pressed the name is changed to the current value in input box and it exits editing mode. If `Escape` is pressed, it only leaves editing mode and does not update name. 
Other functions are same as explained earlier.